scripts/mh_hold_mode.pl

```diff
@@ -0,0 +1,475 @@
+	my $window     = config_window_get($windowrec);
+	my $showstatus = 1;
+
+	if (defined($data))
+	{
+		$data = trim_space($data);
+
+		if ($data ne '')
+		{
+
+			if ($data =~ m/^toggle$/i)
+			{
+				$data = (($window->{'hold_mode'}) ? 'off' : 'on' );
+			}
+
+			if ($data =~ m/^on$/i)
+			{
+				$window->{'hold_mode'} = 1;
+				$windowrec->view()->set_bookmark_bottom('mh_hold_mode');
+				window_refresh();
+
+			} elsif ($data =~ m/^off$/i)
+         	{
+            	$window->{'hold_mode'} = 0;
+				$windowrec->view()->set_bookmark_bottom('mh_hold_mode');
+				window_refresh();
+
+			} elsif ($data =~ m/^flush$/i)
+			{
+				$showstatus = 0;
+				$windowrec->view()->set_bookmark_bottom('mh_hold_mode');
+				window_refresh();
+
+			} elsif ($data =~ m/^scroll_always(.*)$/i)
+			{
+				$showstatus = 0;
+
+				if (defined($1))
+				{
+					$data = trim_space($1);
+
+					if ($data =~ m/^toggle$/i)
+					{
+						$data = (($window->{'scroll_always'}) ? 'off' : 'on' );
+					}
+
+					if ($data =~ m/^on$/i)
+					{
+						$window->{'scroll_always'} = 1;
+
+					} elsif ($data =~ m/^off$/i)
+					{
+						$window->{'scroll_always'} = 0;
+					}
+				}
+
+				$windowrec->print('hold_mode (' . (($window->{'hold_mode'}) ? 'on' : 'off' ) . '): scroll_always is ' . (($window->{'scroll_always'}) ? 'on' : 'off' ), MSGLEVEL_CLIENTCRAP);
+			}
+		}
+	}
+
+	if ($showstatus)
+	{
+		$windowrec->print('hold_mode is ' . (($window->{'hold_mode'}) ? 'on' : 'off' ), MSGLEVEL_CLIENTCRAP);
+	}
+}
+
+##############################################################################
+#
+# statusbar item handlers
+#
+##############################################################################
+
+sub statusbar_more
+{
+	my ($statusbaritem, $get_size_only) = @_;
+
+	my $window = Irssi::active_win();
+
+	my $local_more = 0;
+
+	$local_more = ($window->view()->{'ypos'} - $window->view()->{'height'} + $window->view()->{'empty_linecount'} + 1);
+	$more       = $local_more;
+
+	if ($more)
+	{
+		$statusbaritem->default_handler($get_size_only, "{sb $more more}", '', 0);
+
+	} else {
+
+		$statusbaritem->default_handler($get_size_only, '', '', 0);
+	}
+}
+
+##############################################################################
+#
+# script on load
+#
+##############################################################################
+
+Irssi::command('^SET SCROLL ON');
+
+Irssi::signal_add_last('window created',   'config_window_get');
+Irssi::signal_add_last('window destroyed', 'config_window_del');
+
+for my $window (Irssi::windows)
+{
+	$window->view()->set_scroll(1);
+	$window->view()->set_bookmark_bottom('mh_hold_mode');
+	config_window_get($window);
+}
+
+Irssi::statusbar_item_register('mh_sbmore', '', 'statusbar_more');
+Irssi::statusbars_recreate_items();
+
+statusbar_more_redraw();
+Irssi::timeout_add(1000, 'statusbar_more_redraw', undef);
+
+Irssi::command('^REDRAW');
+
+Irssi::signal_add_first('gui key pressed', 'signal_gui_key_pressed_first');
```

Comment:
I have just started going over it again (old script ive had lying around sincew 2007), so it was only cleaned up a little - didnt wanna mess with the internals till im a bit more into the code again. it does work as it is now, but i can look into using "key send_line" in future versions